internal/v1/handler_blueprints.go

```diff
@@ -300,6 +300,23 @@ func (h *Handlers) UpdateBlueprint(ctx echo.Context, blueprintId uuid.UUID) erro
 		})
 	}
 
+	if blueprintRequest.Customizations.Users != nil {
+		for _, user := range *blueprintRequest.Customizations.Users {
+			// Make sure every user has either ssh key or password set
+			if user.Password == nil && user.SshKey == nil {
+				return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+					Errors: []HTTPError{{
+						Title:  "Invalid user",
+						Detail: "User must have either a password or an SSH key set.",
+					}},
+				})
```

Comment:
I definitely don't insist, but do you think it's worth starting a `validateBlueprint` function?
IMO it then could be shared for Create/Update.

Reply:
Yep I want to, since it wasn't trivial change my logic was to

1. draft tests first
2. make functional version
3. make it pretty (the validation function sounds great, maybe even separate one for customization validation)

and now I just pushed second step :)